fem/integ/bilininteg_trace_jump_ea.cpp

```diff
@@ -0,0 +1,158 @@
+   const int dim = mesh.Dimension();
+   const FaceType ftype = FaceType::Interior;
+   const int nf = mesh.GetNFbyType(ftype);
+
+   const Geometry::Type geom = mesh.GetFaceGeometry(0);
+   const int trial_order = trial_fes.GetMaxElementOrder();
+   const int test_order = test_fes.GetMaxElementOrder();
+   const int qorder = test_order + trial_order - 1;
+   const IntegrationRule &ir = IntRule ? *IntRule : IntRules.Get(geom, qorder);
+   const int nquad = ir.Size();
+
+   Vector pa_data(nquad * nf);
+   {
+      const auto d_w = ir.GetWeights().Read();
+      auto d_pa_data = Reshape(pa_data.Write(), nquad, nf);
+      mfem::forall(nquad * nf, [=] MFEM_HOST_DEVICE (int idx)
+      {
+         const int q = idx % nquad;
+         const int f = idx / nquad;
+         d_pa_data(q, f) = d_w[q];
+      });
+   }
+
+   const FiniteElement &trial_face_el = *trial_fes.GetFaceElement(0);
+   const auto maps = &trial_face_el.GetDofToQuad(ir, DofToQuad::TENSOR);
+   const int ndof_face = trial_face_el.GetDof();
+
+   const Array<real_t> &B = maps->B;
+   const int d1d = maps->ndof;
+   const int q1d = maps->nqpt;
+
+   Vector mass_emat(ndof_face*ndof_face*nf);
+
+   // Note: dim is the element dimension, and we integrate over the faces (one
+   // dimension less)
+   if (dim == 2)
+   {
+      internal::EAMassAssemble1D(nf, B, pa_data, mass_emat, false, d1d, q1d);
+   }
+   else if (dim == 3)
+   {
+      internal::EAMassAssemble2D(nf, B, pa_data, mass_emat, false, d1d, q1d);
+   }
+   else
+   {
+      MFEM_ABORT("Unknown kernel.");
+   }
+
+   const FiniteElement &test_el = *test_fes.GetFE(0);
+   const int n_faces_per_el = 2*dim; // assuming tensor product
+   // Get all the local face maps (mapping from lexicographic face index to
+   // lexicographic volume index, depending on the local face index).
+   Array<int> face_maps(ndof_face * n_faces_per_el);
+   for (int lf_i = 0; lf_i < n_faces_per_el; ++lf_i)
+   {
+      Array<int> face_map(ndof_face);
+      test_el.GetFaceMap(lf_i, face_map);
+      for (int i = 0; i < ndof_face; ++i)
+      {
+         face_maps[i + lf_i*ndof_face] = face_map[i];
+      }
+   }
+
+   Array<int> face_info(nf * 4);
+   {
+      int fidx = 0;
+      for (int f = 0; f < mesh.GetNumFaces(); ++f)
+      {
+         Mesh::FaceInformation finfo = mesh.GetFaceInformation(f);
+         if (!finfo.IsInterior()) { continue; }
+         face_info[0 + fidx*4] = finfo.element[0].local_face_id;
+         face_info[1 + fidx*4] = finfo.element[0].orientation;
+         face_info[2 + fidx*4] = finfo.element[1].local_face_id;
+         face_info[3 + fidx*4] = finfo.element[1].orientation;
+         fidx++;
+      }
+   }
+
+   const int ndof_vol = test_el.GetDof();
+   const auto d_face_maps = Reshape(face_maps.Read(), ndof_face, n_faces_per_el);
+   const auto d_face_info = Reshape(face_info.Read(), 2, 2, nf);
+
+   real_t *d_emat;
+   if (add)
+   {
+      d_emat = emat.ReadWrite();
+   }
+   if (!add)
```

Comment:
(nit) else? 

```suggestion
   else
```

Reply:
Fixed in 4979dc1701e2b9d1ac3159ff55bdf4155fb5654e.